### DISCUSSION TITLE
Spore Meeting Agenda @ 2024-01-09

### DISCUSSION THREAD

**Discussion by ashuralyk** [sporeprotocol/spore-scrum#12] [Show and tell]
# Goal
[[previous]](https://github.com/sporeprotocol/spore-scrum/discussions/11)

1. complete the integration of co-build into `spore-sdk` as fast as possible
2. integrate CI workflow of `spore-sdk` into `spore-contract` repo
3. complete the final work of the standalone CI workflow of `spore-sdk`
4. solve bugs in both of sdk and contracts

p.s. after discussion, version `v2` has been chosen to launch the mainnet publish event

# Previous Work

1. integration of co-build
    - [x] integrate co-build into `spore-contract` https://github.com/sporeprotocol/spore-contract/pull/28
    - [x] start to learn how to integrate co-build into `spore-sdk`

2. bugs of the integration of version `v2`
    - [x] https://github.com/sporeprotocol/spore-sdk/pull/51
    - [x] https://github.com/sporeprotocol/spore-sdk/pull/60
    - [ ] https://github.com/sporeprotocol/spore-contract/pull/32

3. preparation of standalone CI workflow of `spore-sdk`
    - [x] rewrite and close previous test cases https://github.com/sporeprotocol/spore-sdk/pull/56 https://github.com/sporeprotocol/spore-sdk/pull/48
    - [x] run CI workflow in Github https://github.com/sporeprotocol/spore-sdk/actions/runs/7452098852/job/20274626103?pr=60

4. 100K CKB has been received, and proposal in Nervos DAO and link that refers to Nervos Talk have been done

# Next Plan

1. complete the integration of co-build, and then start to test it in CI workflow
2. copy the entire CI workflow of `spore-sdk` into `spore-contract`, and then each change of contract can be checked immediately
3. review the entire code base of `spore-contract`, then find and solve bugs as much as possible
4. start e2e test on `spore-demo`

--------

cc @RongxinZhang @ShookLyngs @Flouse @Dawn-githup

**Comment by Dawn-githup** [comment]
Done:
  
  1. Simplify the spore-dev process: https://github.com/Dawn-githup/spore_devenv
  2. spore-sdk #51 test：[fix(sdk): fix ClusterProxyArgs.minPayment type spore-sdk#51](https://github.com/sporeprotocol/spore-sdk/pull/51)
  3. spore-sdk #55 review: https://github.com/sporeprotocol/spore-sdk/pull/55 
  
Next:

  1. spore-demo e2e functional test: https://mubu.com/doc/46ygy0TIl3M
  2. spore-dev optimizes details and adds other features：https://github.com/cryptape/acceptance-internal/issues/619

**Comment by ShookLyngs** [comment]
Done:
1. Automated sdk tests: https://github.com/sporeprotocol/spore-sdk/pull/56
2. Optimized sdk tests and fixed bugs: https://github.com/sporeprotocol/spore-sdk/pull/60
3. Removed a mime package from the deps: https://github.com/sporeprotocol/spore-sdk/pull/61
4. Intergrated devnet testing of the sdk with dawn: https://github.com/sporeprotocol/spore-sdk/pull/62

Next:
1. Co-build intergration: https://github.com/sporeprotocol/spore-sdk/issues/53
2. Adapt ClusterProxy logic changes: https://github.com/sporeprotocol/spore-sdk/issues/57

**Comment by ashuralyk** [comment]
# Progress of Scrum
1. CI workflow of `spore-sdk` is already running in Github CI of `spore-contract`
2. `spore-demo` is started to update to catch latest version of contract and sdk, but it needs the update of sdk to go next step
3. `spore-sd` has been working on the integration of co-build
